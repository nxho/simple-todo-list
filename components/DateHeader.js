import React from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';

export default class DateHeader extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      date: new Date(),
    }
  }

  formatMonth(date) {
    return new Intl.DateTimeFormat('en-US', { month: 'long' }).format(date).substr(0,3).toUpperCase();
  }

  formatWeekday(date) {
    return new Intl.DateTimeFormat('en-US', { weekday: 'long' }).format(date).toUpperCase();
  }

  render() {
    const { date } = this.state;

    return(
      <View style={styles.container}>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.dayText} testID='day'>
            { date.getDate() }
          </Text>
          <View style={{ justifyContent: 'center', marginLeft: 5 }}>
            <Text style={styles.monthText} testID='month'>
              { this.formatMonth(date) }
            </Text>
            <Text style={styles.yearText} testID='year'>
              { date.getFullYear() }
            </Text>
          </View>
        </View>
        <View style={{ justifyContent: 'center' }}>
          <Text style={styles.weekdayText} testID='weekday'>
            { this.formatWeekday(date) }
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  dayText: {
    fontSize: 45,
  },
  monthText: {
    fontWeight: 'bold',
  },
  yearText: {
  },
  weekdayText: {
  },
});

