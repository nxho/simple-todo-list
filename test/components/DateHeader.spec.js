import React from 'react';
import { render, shallow, fireEvent } from 'react-native-testing-library';
import DateHeader from '../../components/DateHeader';

describe('DateHeader component', () => {
  const origDate = Date;
  const testDate = new Date(1549411200000);

  beforeEach(() => {
    // mock current date to Wednesday, February 5, 2019, 12:00:00 AM
    global.Date = class extends Date {
      constructor() {
        return testDate;
      }
    }
  });

  afterEach(() => {
    global.Date = origDate;
  });

  it('renders correctly', () => {
    const { output } = shallow(<DateHeader />);

    expect(output).toMatchSnapshot();
  });

  it('displays current date correctly', () => {
    const { getByTestId } = render(<DateHeader />);

    expect(getByTestId('month').props.children).toEqual(new Intl.DateTimeFormat('en-US', { month: 'long' }).format(testDate).substr(0,3).toUpperCase());
    expect(getByTestId('day').props.children).toEqual(testDate.getDate());
    expect(getByTestId('year').props.children).toEqual(testDate.getFullYear());
    expect(getByTestId('weekday').props.children).toEqual(new Intl.DateTimeFormat('en-US', { weekday: 'long' }).format(testDate).toUpperCase());
  });
});

