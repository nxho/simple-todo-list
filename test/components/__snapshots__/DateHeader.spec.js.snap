// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader component renders correctly 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "fontSize": 45,
        }
      }
      testID="day"
    >
      5
    </Component>
    <Component
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <Component
        style={
          Object {
            "fontWeight": "bold",
          }
        }
        testID="month"
      >
        FEB
      </Component>
      <Component
        style={Object {}}
        testID="year"
      >
        2019
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Component
      style={Object {}}
      testID="weekday"
    >
      TUESDAY
    </Component>
  </Component>
</Component>
`;
